libs/cardano-ledger-core/src/Cardano/Ledger/UMap.hs

```diff
@@ -271,539 +398,578 @@ toUMapPair (UMap !m1 !m2) =
 --
--- Replaces the value already in the ViewMap with 'v', if key 'k' is already there:
+--   1. to keep the old value
+--   > insertWith' (\ old new -> old) k v view
 --
--- > insertWith' (\ old new -> new) k v xs
+--   2. to replace the old value with the new value
+--   > insertWith' (\ old new -> new) k v view
 --
--- Replaces the value already in the ViewMap with the sum, if key 'k' is already there:
+--   3. to combine the old and new values with summation
+--   > insertWith' (\ old new -> old + new) k v view
 --
--- > insertWith' (\ old new -> old+new) k v xs
---
--- Ignores 'combine' if the key 'k' is NOT already in the ViewMap, and inserts 'v':
---
--- > insertWith' combine k v xs
-insertWith' ::
-  (v -> v -> v) ->
-  k ->
-  v ->
-  View c k v ->
-  View c k v
-insertWith' comb stakeid newpair (RewardDeposits (UMap tripmap ptrmap)) =
-  rdPairs (Map.alter comb2 stakeid tripmap) ptrmap
-  where
-    -- Here 'v' is (CompactForm Coin), but the UMap stores Word64,
-    -- so there is some implict coercion going on here using the Triple pattern
-    comb2 Nothing = zeroMaybe (Triple (SJust newpair) Set.empty SNothing)
-    comb2 (Just (Triple (SJust oldpair) x y)) = zeroMaybe (Triple (SJust (comb oldpair newpair)) x y)
-    comb2 (Just (Triple SNothing x y)) = zeroMaybe (Triple (SJust newpair) x y)
-insertWith' comb stakeid newpoolid (Delegations (UMap tripmap ptrmap)) =
-  delegations (Map.alter comb2 stakeid tripmap) ptrmap
-  where
-    comb2 Nothing = Just (Triple SNothing Set.empty (SJust newpoolid))
-    comb2 (Just (Triple x y (SJust old))) = Just (Triple x y (SJust (comb old newpoolid)))
-    comb2 (Just (Triple x y SNothing)) = Just (Triple x y (SJust newpoolid))
-insertWith' comb ptr stake (Ptrs (UMap tripmap ptrmap)) =
-  let (oldstake, newstake) =
-        case Map.lookup ptr ptrmap of -- This is tricky, because we need to retract the oldstake
-          Nothing -> (stake, stake) -- and to add the newstake to maintain the UMap invariant
-          Just stake2 -> (stake2, comb stake2 stake)
-      -- Delete old pointer from set in Triple, but also delete the whole triple if it goes to Zero.
-      retract stakeid pointer m = Map.update ok stakeid m
-        where
-          ok (Triple c set d) = zeroMaybe (Triple c (Set.delete pointer set) d)
-      -- Add the new pointer to the set in Triple
-      tripmap2 = Map.alter addPtr newstake (retract oldstake ptr tripmap)
+-- If `k` does not exist as a key in the `UView`,
+--   the combining function is ignored, and
+--   the key `k` and the value `v` are inserted into the map-like `UView`
+--   > insertWith' ignoredCombiningFunction k v view
+insertWith' :: (v -> v -> v) -> k -> v -> UView c k v -> UView c k v
+insertWith' combine key val = \case
+  RewDepUView UMap {..} -> rewDepUView (Map.alter go key umElem) umPtr
+    where
+      -- \| Here 'val' is (CompactForm Coin), but the UMap stores Word64,
+      -- so there is some implict coercion going on here using the UElemP pattern
```

Comment:
You can't add haddock for non-top level bindings. That's why fourmolu changes `|` -> `\|` and `^` -> `\^`. So you might wanna do a grep remove of those. Unless you like to have those there, in which case feel free to leave 'em as is
```suggestion
      -- Here 'val' is (CompactForm Coin), but the UMap stores Word64,
      -- so there is some implict coercion going on here using the UElemP pattern
```

Reply:
Completely makes sense. Thanks! I didn't notice this earlier. Removing all the non-compliant haddock.